File: binary_search.java
//non method
import java.util.Scanner;
import  static java.lang.System.*;

public class binary_search {
    public static void main(String[] args) {
     Scanner scan = new Scanner(System.in);
     out.println("please insert your integer of choice :");
     int num = scan.nextInt();
     
     //creating array
     int list [] = new int[5];
     for (int i = 0;i < list.length;i++){
      out.println("insert value for dataset :");
      list[i] = scan.nextInt();
     }
     //binary search 
    for (int j = 0;j < 1;j++){
        int min = 0;
        int max  = list.length;
        int mid = (max + min)/2;
        int number = 0;
        for (int k = 0;k < list.length;k++){
            if (num == list[k]){
                out.println("value "+ num + " found at index "+ k );
                number = num;
            }
            else{
                if (num > mid){
                    min = mid;
                    mid = (max + min)/2;
                }else{
                    max = mid;
                    mid = (max + min)/2;
                }

            }//end of if
        }
        
        if (num !=number )
        out.print("does not exist");
    }
    


    }
    
}

File: binary_search.py


def binary(input,list):
    min = 0 
    max = len(list)
    mid = 0
    for num,x in enumerate (list):
        if input > x:
            if x == input :
                return f"value of {x} found at index {num}"         
    
            else:
                min = x
                mid = (max+min)/2
        else :
            if x == input :
                return f"value of {x} found at index {num}"         
    
            else:
                max = x
                mid = (max+min)/2
    return f"value of {input} not found"         
    
print(binary(90,[1,33,44,66,90,99]))
print(binary(990,[22,33,900,990,100]))
print(binary(910,[22,33,900,990,100000]))
print(binary(100000,[22,33,900,990,100000]))


 
